src/Servers/HttpSys/src/RequestProcessing/Response.cs

```diff
@@ -403,7 +404,12 @@ internal uint ComputeHeaders(long writeCount, bool endOfRequest = false)
 
         // Determine if the connection will be kept alive or closed.
         var keepConnectionAlive = true;
-        if (requestVersion <= Constants.V1_0 // Http.Sys does not support "Keep-Alive: true" or "Connection: Keep-Alive"
+
+        // An HTTP/1.1 server may also establish persistent connections with
+        // HTTP / 1.0 clients upon receipt of a Keep-Alive connection token.
+        // However, a persistent connection with an HTTP/1.0 client cannot make
+        // use of the chunked transfer-coding. From: https://www.rfc-editor.org/rfc/rfc2068#section-19.7.1
+        if ((requestVersion <= Constants.V1_0 && (!requestConnectionKeepAliveSet || responseChunkedSet))
```

Comment:
I didn't know the bit about HTTP/1.0 clients not being allowed to use chunked transfer encoding on HTTP/1.1 keep-alive connections. The updated sections in [RFC 2616](https://www.rfc-editor.org/rfc/rfc2616#section-19.6.2) and [RFC 9112](https://www.rfc-editor.org/rfc/rfc9112#appendix-C.2.2) seem to be less opinionated, but they reference RFC 2068, so it seems like a good idea to play it safe and close the connection in the "responseChunkedSet" case.

I think we should do the same thing in Kestrel:

https://github.com/dotnet/aspnetcore/blob/59d4050e7a26326764aa0f008b378f0adbc40efa/src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs#L1145-L1155

We could add the following and a test to keep the behaviors of Http.Sys and Kestrel aligned:

```csharp
if (hasTransferEncoding && _httpVersion == Http.HttpVersion.Http10)
{
    _keepAlive = false;
}
```

It might also be a good idea to have HttpSys do the `HttpHeaders.GetFinalTransferCoding(responseHeaders.HeaderTransferEncoding) != TransferCoding.Chunked` check that Kestrel does and disable keep-alive connections even for HTTP/1.1 due to RFC 7230 3.3.1.

I also don't think `var responseChunkedSet = Matches(Constants.Chunked, transferEncodingString);` properly handles things like apps that manually set something like `Transfer-Encoding: gzip, chunked`, so it's probably good to call `HttpHeaders.GetFinalTransferCoding` just to fix the other logic that relies on `responseChunkedSet`.

Reply:
Moved HttpHeaders.GetFinalTransferCoding into StringUtilities as a shared source between the projects. And fix the other logic that relies on responseChunkedSet. I'll leave the Kestrel changes to a separate PR.